crates/ironrdp-client/src/rdp.rs

```diff
@@ -178,6 +192,41 @@ async fn active_session(
                         // TODO: https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/27915739-8f77-487e-9927-55008af7fd68
                         break 'outer;
                     }
+                    RdpInputEvent::Clipboard(event) => {
+                        if let Some(cliprdr) = active_stage.get_svc::<ironrdp::cliprdr::Cliprdr>() {
+                            let svc_request = match event {
+                                ClipboardMessage::SendInitiateCopy(formats) => {
+                                    let request = cliprdr.initiate_copy(&formats)
+                                        .map_err(|e| session::custom_err!("CLIPRDR", e))?;
+                                    Some(request)
+                                }
+                                ClipboardMessage::SendFormatData(response) => {
+                                    let request = cliprdr.sumbit_format_data(response)
+                                        .map_err(|e| session::custom_err!("CLIPRDR", e))?;
+                                    Some(request)
+                                }
+                                ClipboardMessage::SendInitiatePaste(format) => {
+                                    let request = cliprdr.initiate_paste(format)
+                                        .map_err(|e| session::custom_err!("CLIPRDR", e))?;
+                                    Some(request)
+                                }
+                                ClipboardMessage::Error(e) => {
+                                    error!("Clipboard backend error: {}", e);
+                                    None
+                                }
+                            };
+
+                            if let Some(request) = svc_request {
+                                let frame = active_stage.process_user_svc_request(request)?;
+                                vec![ActiveStageOutput::ResponseFrame(frame)]
+                            } else {
+                                vec![]
```

Comment:
Very much a nit: I would recommend `Vec::new()` to avoid the useless macro expansion, but I don’t think it matters that much (feel free to ignore).
```suggestion
                                Vec::new()
```
(Also, everywhere else `Vec::new()` is used)